azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java

```diff
@@ -123,7 +138,18 @@ public long getQueuedFlowSize() {
    */
   @Override
   public DispatchMethod getDispatchMethod() {
-    return DispatchMethod.CONTAINERIZED;
+    if (this.rampUp == 0) {
```

Comment:
DispatchMethod is very useful and it controlling the behaviour at the moment by turning off and turning on feature. DispatchMethod is currently specified through configs. Hence, dispatch method is initialized during web server startup. The containerization feature is turned off if dispath method is POLL. If dispatch method is set to CONTAINERIZED then only all the guava juice binding for containerization are initialized. As we are dynamically toggling between  POLL and CONTAINERIZED based on rampup we need to reload the bindings as well. In other words, we need to always intialized the bindings or look at different way of handling this.

Reply:
+1. I didn't see any changes related to injection over here. How will implementation for POLL method be injected in case of ramp up?

Reply:
Got it. And specifying dispatch method as POLL from executor side will take care of polling executions. Thank you for clarifying this.